static/src/javascripts/bootstraps/enhanced/media/main.js

```diff
@@ -185,6 +185,7 @@ define([
             embedPath = $el.attr('data-embed-path'),
             // we need to look up the embedPath for main media videos
             canonicalUrl = $el.attr('data-canonical-url') || (embedPath ? '/' + embedPath : null),
+            shouldHideAdverts = $el.attr('data-block-video-ads') === 'false' ? false : true,
```

Comment:
Order:

* has attribute? read attribute => set `shouldHideAdverts`
* has canonical URL? look it up via `info.json` => set `shouldHideAdverts`

So we should be right nearly all of the time except, like the other day, the `info.json` borks.